pkg/container-utils/containerd/containerd.go

```diff
@@ -15,31 +15,246 @@
+	containerID, err := runtimeclient.ParseContainerID(runtimeclient.ContainerdName, containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	containerData, _, _, err := c.getContainerDataAndContainerAndTask(containerID)
+	if err != nil {
+		return nil, err
+	}
+	return containerData, nil
+}
+
+func (c *ContainerdClient) GetContainerDetails(containerID string) (*runtimeclient.ContainerDetailsData, error) {
+	containerID, err := runtimeclient.ParseContainerID(runtimeclient.ContainerdName, containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	containerData, container, task, err := c.getContainerDataAndContainerAndTask(containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	spec, err := container.Spec(context.TODO())
+	if err != nil {
+		return nil, fmt.Errorf("getting spec for container %q: %w", containerID, err)
+	}
+
+	mountData := make([]runtimeclient.ContainerMountData, len(spec.Mounts))
+	for i := range spec.Mounts {
+		mount := spec.Mounts[i]
+		mountData[i] = runtimeclient.ContainerMountData{
+			Source:      mount.Source,
+			Destination: mount.Destination,
+		}
+	}
+
+	return &runtimeclient.ContainerDetailsData{
+		ContainerData: *containerData,
+		Pid:           int(task.Pid()),
+		CgroupsPath:   spec.Linux.CgroupsPath,
+		Mounts:        mountData,
 	}, nil
 }
+
+func (c *ContainerdClient) getContainerDataAndContainerAndTask(containerID string) (*runtimeclient.ContainerData, containerd.Container, containerd.Task, error) {
+	container, err := c.getContainerdContainer(containerID)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	task, err := getContainerdTask(container)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	containerData, err := taskAndContainerToContainerData(task, container)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return containerData, container, task, nil
+}
+
+// getContainerdContainer returns the corresponding container.Container instance to
+// the given id
+func (c *ContainerdClient) getContainerdContainer(id string) (containerd.Container, error) {
+	container, err := c.client.LoadContainer(context.TODO(), id)
+	if err != nil {
+		return nil, fmt.Errorf("loading container with id %q: %w", id, err)
+	}
+
+	if isSandboxContainer(container) {
+		log.Debugf("ContainerdClient: container %q is a sandbox container. Temporary skipping it", container.ID())
+		return nil, runtimeclient.ErrPauseContainer
+	}
+
+	return container, nil
+}
+
+// getContainerdTask returns the corresponding container.Task instance to
+// the given containerd.Container. If there is no task, nil is returned without an error
+func getContainerdTask(container containerd.Container) (containerd.Task, error) {
+	task, err := container.Task(context.TODO(), nil)
+	if err != nil {
+		if !errdefs.IsNotFound(err) {
+			return nil, fmt.Errorf("getting task for container %q: %w", container.ID(), err)
+		}
+		// else it is created, but not running
+		log.Debugf("No task for %q", container.ID())
```

Comment:
Could you please add the prefix `ContainerdClient` to all logs?

```suggestion
		log.Debugf("ContainerdClient: No task for %q", container.ID())
```